app/controllers/verify/phone_controller.rb

```diff
@@ -1,22 +1,50 @@
 module Verify
   class PhoneController < StepController
+    before_action :confirm_step_needed
+
     helper_method :idv_phone_form
+    helper_method :phone_confirmation_failed?
 
     def new
       analytics.track_event(Analytics::IDV_PHONE_RECORD_VISIT)
     end
 
     def create
-      if idv_phone_form.submit(phone_params)
+      if idv_phone_form.submit(phone_params) && confirm_phone_with_vendor?
```

Comment:
Same comment here about adding this logic as a new validation in `Idv::PhoneForm`.